crates/swc_ecma_minifier/src/analyzer/storage/normal.rs

```diff
@@ -98,8 +101,10 @@ impl Storage for ProgramData {
 
                     match kind {
                         ScopeKind::Fn => {
-                            e.get_mut().is_fn_local = false;
-                            e.get_mut().used_by_nested_fn = true;
+                            if !var_info.used_recursively {
+                                e.get_mut().is_fn_local = false;
```

Comment:
I've been thinking about this for a while

Reply:
I found that this is problematic for some cases.


In the code below, 

```js

var foo;
function init() {
    function bar() {
        foo = bar;
    }

    escape(bar)
}

```

`bar` is not fn-local but it wil be marked as fn-local

Reply:
I expect something similar to this pattern to exist in real-world apps, so I think we should fix it

Reply:
I think tracking the count of non-fn-local usage can be a way

Reply:
But for 

```js
function a() {
  x(a)
}
module.export = a
```

I think it's better to extend sequential inliner.

Reply:
Oh, you're right. Let me check how terser deal with this when I have time.

Reply:
Your example is wrong on [playground](https://play.swc.rs/?version=1.3.3&code=H4sIAAAAAAAAA0utKMgvKlEoSyxSSMvPt%2BZKhfDTSvOSSzLz8xQy8zJLNDQVqrngIkmJRWABBSAAalGwBYlYg7nJ%2BXnF%2BTmpejn56RqGhoaaXLVgYbAOIBsARRRWE2wAAAA%3D&config=H4sIAAAAAAAAA21TQZLjIAz8i8972MpxHrC3fQNFQDiaxcgliUxcU%2Fn7iCTGcbI31O6WRav5Hj4lDB%2Ffw%2BxZgNtJlqL%2BMnwMuswggXHW4degYlDyWeBqBfsiiXhqfAYf9HaoRXECU%2FqqNHnFMFyNnYkEHmLTeh7B%2BAPI4ffhYK0nLJiW1iHQNDOItLPR6gRFpSs9M31ZqVytOhJlsDFea%2BfFYVEYgTdpoJz9LODOnrui%2Fc0zCpUdVBWim5nmJ7REVKRizVYsgo8uUIQOIENQPMNGsR5GKWKD9UluYIRjHcdm950JZ5%2Br194LLjcb7I9ddyIUdakWeYF2k96h%2ByUfLEyOQSuXlfNJWHY%2B%2FAOwObMXKX6CTXnDk%2B1hz0z%2FYWFJtkRdOmo736YqMNrlHGLqN2xTAytubjHEGqDdMGzNH%2BDuQoIRHKRkbnexfKGG09asRZdSL801n7Yd3Eu35mmPtgi8gX9sKn1exwO3kJ9eMVmmI%2BW3FhPoieIbbEYovYJs6b3Mr2gtEcxmiE8fqrSyx9SWr%2BRye1%2BrzxYI07gx07HH99qe5frs1pzIX4o1PzZ0%2FQEAk%2FXaGQQAAA%3D%3D)-- let me try to fix it first.

Reply:
Current behavior is aligned with terser.
```js
export var foo;
export function init() {
    function bar() {
        foo = bar;
        console.log(111);
    }
    return bar;
}
```
would inline, while
```
export var foo;

function bar() {
    foo = bar;
    console.log(111);
}
export function init() {
    return bar;
}
```
would not. I believe this is fine.

Reply:
Can you add a test for

```js

export function foo() {
    function bar() {
        function baz() {
            return bar;
        }
        return bar && baz;
    }

    return bar;
}

```

?
I think in this case `bar` should not be `fn-local`, but it will not be marked as non-fn-local because of recursion

Reply:
In this case I see no reason why `bar` should not be inlined.

Reply:
It's fine if fn-local is used only for inlining, but not sure if it's the case

Reply:
I checked the code and `fn-local` is used by much more places. So I think we should analyze it correctly, or use another field like `fn_local_for_inlining` just for inlining